src/jit/gentree.cpp

```diff
@@ -11917,7 +11930,7 @@ GenTree* Compiler::gtFoldExprCall(GenTreeCall* call)
 
     if (ni == NI_System_Enum_HasFlag)
     {
-        GenTree* thisOp = call->gtCallObjp;
+        GenTree* thisOp = call->gtCallThisArg->GetNode();
```

Comment:
It could but I would hope that we can get rid of `gtCallThisArg` completely in the future. I don't see any good reason for `this` arg to not be part of the `gtCallArgs` list.